osquery/core/watcher.cpp

```diff
@@ -287,19 +287,31 @@ void WatcherRunner::watchExtensions() {
 
       // The extension manager also watches for extension-related failures.
       // The watchdog is more general, but may find failed extensions first.
-      createExtension(extension.first);
-      extension_restarts_[extension.first] += 1;
+      if (extension_backoff_[extension.first] == 0) {
+        createExtension(extension.first);
+        ++extension_restarts_[extension.first];
+        extension_backoff_[extension.first] =
+            2 * extension_restarts_[extension.first];
+        VLOG(1) << extension.first << " back off set to "
+                << extension_backoff_[extension.first];
+      } else {
+        VLOG(1) << "Waiting for backoff before restarting " << extension.first;
+        --extension_backoff_[extension.first];
+      }
     } else {
-      extension_restarts_[extension.first] = 0;
+      if (extension_restarts_[extension.first] > 0) {
+        extension_restarts_[extension.first] = 0;
+        VLOG(1) << "Setting backoff for " << extension.first << " to 0";
+      }
     }
   }
   // If any extension creations failed, stop managing them.
-  for (auto& extension : extension_restarts_) {
+  /*for (auto& extension : extension_restarts_) {
```

Comment:
I don't know if the backoff is needed. I would recommend just removing this `for` loop in this commit and calling this issue fixed.